pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,465 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the MPLDrawer class for creating circuit diagrams with matplotlib
+"""
+from collections.abc import Iterable
+
+import matplotlib.pyplot as plt
+from matplotlib import patches
+
+
+def _to_tuple(a):
+    """converts int or iterable to always tuple"""
+    if isinstance(a, Iterable):
+        return tuple(a)
+    return (a,)
+
+
+class MPLDrawer:
+    r"""Allows easy creation of graphics representing circuits with Matplotlib
+
+    Args:
+        n_layers (Int): the number of layers
+        n_wires (Int): the number of wires
+
+    Keyword Args:
+        figsize=None (Iterable): Allows user's to manually specify the size of the figure.  Defaults
+           to scale with the size of the circuit via ``n_layers`` and ``n_wires``.
+
+    **Example**
+
+    .. code-block:: python
+
+        drawer = MPLDrawer(n_wires=5,n_layers=5)
+
+        drawer.label(["0","a",r"$|\Psi\rangle$",r"$|\theta\rangle$", "aux"])
+
+        drawer.box_gate(0, [0,1,2,3,4], "Entangling Layers", rotate_text=True)
+        drawer.box_gate(1, [0, 1], "U(θ)")
+        drawer.box_gate(1, 4, "X", color='lightcoral')
+
+        drawer.SWAP(1, (2, 3))
+        drawer.CNOT(2, (0,2), color='forestgreen')
+
+        drawer.ctrl(3, [1,3])
+        drawer.box_gate(3, 2, "H", zorder_base=2)
+
+        drawer.ctrl(4, [1,2])
+
+        drawer.measure(5, 0)
+
+        drawer.fig.suptitle('My Circuit', fontsize='xx-large')
+
+    .. figure:: ../../_static/drawer/example_basic.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+    .. UsageDetails::
+
+    This class uses matplotlib and pyplot.  The figure and axes objects can be accessed via ``drawer.fig``
+    and ``drawer.ax`` respectively for further configuration. For example, the above example circuit manipulates the
+    pyplot figure to set a title using ``drawer.fig.suptitle``. You can also save the images using ``plt.savefig``.
+
+    Each gate takes arguments in order of ``layer`` followed by ``wires``.  These translate to ``x`` and
+    ``y`` coordinates in the graph. Layer number (``x``) increases as you go right, and wire number
```

Comment:
I would strongly recommend this to be reversed. The qubit-ordering convention in PennyLane is that in a tensor product, qubits are numbered as `|0, 1, 2, ...>`, which corresponds to a list of wire labels `[0, 1, 2,...]`, and to a circuit diagram in which wire 0 is at the top and increases as you go down (furthermore `qml.draw` orders the labels passed in `wire_order` from top to bottom).

Reply:
Added `self.ax.invert_yaxis()`. This way I don't have to do any coordinate transformations :)